protocols/directfund/directfund.go

```diff
@@ -101,6 +102,13 @@ func New(
 
 // Public methods on the DirectFundingObjectiveState
 
+// MarshalJSON returns a JSON representation of the DirectFundObjective
+// with channel state replaced by a channel Id
+func (s DirectFundObjective) MarshalJSON() ([]byte, error) {
+	jsonDFO := JSONDirectFundObjective{s, s.C.Id}
```

Comment:
Why not make the `JSON*` struct types private to _this method_ ?

Reply:
I suppose because you can reuse them when un-marshaling. Seems like the right tradeoff, probably.